pdns/ws-auth.cc

```diff
@@ -591,6 +580,208 @@ static void apiZoneCryptokeys(HttpRequest* req, HttpResponse* resp) {
     throw HttpNotFoundException();
   }
   resp->setBody(doc);
+
+}
+
+/*
+ * This method handles DELETE requests for URL /api/v1/servers/:server_id/zones/:zone_name/cryptokeys/:cryptokey_id .
+ * It deletes a key from :zone_name specified by :cryptokey_id.
+ * Server Answers:
+ * Case 1: zone_name not found
+ *      The server returns 400 Bad Request
+ * Case 2: the backend returns true on removal. This means the key is gone.
+ *      The server returns 200 No Content
+ * Case 3: the backend returns false on removal. An error occoured.
+ *      The sever returns 422 Unprocessable Entity with message "Could not DELETE :cryptokey_id"
+ * */
+static void apiZoneCryptokeysDELETE(DNSName zonename, int inquireKeyId, HttpRequest *req, HttpResponse *resp) {
+  if (!req->parameters.count("key_id"))
```

Comment:
why do you have `inquireKeyId` in the param list and then check for `key_id` in req->parameters too?